pycbc/events/cuts.py

```diff
@@ -0,0 +1,347 @@
+                        help="Cuts to apply to the triggers, supplied as "
+                             "PARAMETER:VALUE:LIMIT. Format is the same as in "
+                             "--trigger-cuts. PARAMETER can be one of '"
+                             + "', '".join(template_param_choices) + "'.")
+
+
+def convert_inputstr(inputstr, choices):
+    """
+    Convert the inputstr into a dictionary keyed on parameter
+    with a tuple of the function to be used in the cut, and
+    the float to compare to.
+    Do input checks
+    """
+    try:
+        cut_param, cut_value_str, cut_limit = inputstr.split(':')
+    except ValueError as value_e:
+        logging.warning("ERROR: Cut string format not correct, please "
+                        "supply as PARAMETER:VALUE:LIMIT")
+        raise value_e
+
+    if cut_param.lower() not in choices:
+        raise NotImplementedError("Cut parameter " + cut_param.lower() + " "
+                                  "not recognised, choose from "
+                                  + ", ".join(choices))
+    if cut_limit.lower() not in ineq_choices:
+        raise NotImplementedError("Cut inequality " + cut_limit.lower() + " "
+                                  "not recognised, choose from "
+                                  + ", ".join(ineq_choices))
+
+    try:
+        cut_value = float(cut_value_str)
+    except ValueError as value_e:
+        logging.warning("Error: Cut value must be convertible into a float, "
+                        "got '%s', a %s.", cut_value, type(cut_value))
+        raise value_e
+
+    return {cut_param: (ineq_functions[cut_limit],
+                        float(cut_value_str))}
+
+
+def ingest_cuts_option_group(args):
+    """
+    Return dictionaries for trigger and template cuts.
+    """
+    # Deal with the case where no cuts are supplied:
+    if not args.trigger_cuts and not args.template_cuts:
+        return {}, {}
+
+    # Deal with the case where one set of cuts is supplied
+    # but not the other
+    trigger_cut_strs = args.trigger_cuts or []
+    template_cut_strs = args.template_cuts or []
+
+    # Handle trigger cuts
+    trigger_cut_dict = {}
+    for inputstr in trigger_cut_strs:
+        trigger_cut_dict.update(convert_inputstr(inputstr,
+                                                 trigger_param_choices))
+    # Handle template cuts
+    template_cut_dict = {}
+    for inputstr in template_cut_strs:
+        template_cut_dict.update(convert_inputstr(inputstr,
+                                                  template_param_choices))
+
+    return trigger_cut_dict, template_cut_dict
+
+
+def apply_trigger_cuts(triggers, trigger_cut_dict):
+    """
+    Fetch/Calculate the parameter for triggers, and then
+    apply the cuts defined in template_cut_dict
+
+    Parameters
+    ----------
+    triggers: ReadByTemplate object
+        The triggers in this particular template. This
+        must have the correct datasets required tio calculate
```

Comment:
typo 'tio'. I'm not sure if the comment about 'okay in standard searches' is going to be informative .. if the function works people won't need the comment, if it doesn't work then it probably won't help with debugging